core/src/main/java/io/grpc/internal/TimeProvider.java

```diff
@@ -27,12 +27,9 @@
   long currentTimeNanos();
 
   TimeProvider SYSTEM_TIME_PROVIDER = new TimeProvider() {
-    final long offsetNanos =
-        TimeUnit.MILLISECONDS.toNanos(System.currentTimeMillis()) - System.nanoTime();
-
     @Override
     public long currentTimeNanos() {
-      return System.nanoTime() + offsetNanos;
+      return TimeUnit.MILLISECONDS.toNanos(System.currentTimeMillis());
     }
```

Comment:
After your change, it will return the same nano value for any invocation that happens within one millisecond.